.github/workflows/testing.yml

```diff
@@ -0,0 +1,37 @@
+name: run test cases
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  testing:
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-20.04]
+        compiler: [gcc, clang]
+
+    runs-on: ${{ matrix.os }}
+
+    env:
+      CC: ${{ matrix.compiler }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: build packcc
+      run: |
+        (
+        cd  build/$CC
+        make
+        )
+    - name: install bats
+      run: |
+        curl -L -o bats-core-1.2.1.tar.gz https://github.com/bats-core/bats-core/archive/v1.2.1.tar.gz
+        tar zxvf bats-core-1.2.1.tar.gz
+        cd bats-core-1.2.1 &&
+        ./install.sh /usr/local
+    - name: run test cases
```

Comment:
If we have `make check` we can remove this.

Reply:
Done.